src/bank.rs

```diff
@@ -35,10 +35,13 @@ pub const MAX_ENTRY_IDS: usize = 1024 * 16;
 pub const VERIFY_BLOCK_SIZE: usize = 16;
 
 /// Reasons a transaction might be rejected.
+#[cfg_attr(feature = "cargo-clippy", allow(large_enum_variant))]
 #[derive(Debug, PartialEq, Eq)]
 pub enum BankError {
     /// Attempt to debit from `Pubkey`, but no found no record of a prior credit.
-    AccountNotFound(Pubkey),
+    /// `Instruction` that was sent in the transaction, and 'Hash' is the last_id
+    /// in the transaction
+    AccountNotFound(Pubkey, Instruction, Hash),
```

Comment:
I'd recommend reverting the changes to AccountNotFound in this PR. It adds a lot of unnecessary risk to developer productivity for both you and others. Refactoring PRs are awesome, but we need to consider the consequences of changing hundreds of lines of code. First, if the PR is hard to review, it means you'll get stuck in rebase hell as we sift through all the changes. Second, and perhaps more importantly, if it breaks something and we want to revert it, that's a big pain in the ass. So when doing PRs like this, we need to keep changes to functionality to an absolute minimum. We want these PRs to be merged quickly after they were submitted for review and never need to be reverted.

This change is particular has already slowed down PR progress considerably and is looking like it'll continue to. Adding that `Instruction` raises 3 red flags in my mind:

* Why would we add non-account information to an error object? Is there any code that would ever index on that data? Would it be sufficient to add that data to a `error!` or `debug!` message instead?
* Does that `instruction.clone()` below introduce a DoS opportunity? A transaction with a valid signature from a non-existent account is one of the simplest ways to spam the system. This change adds an copy of the transaction's largest field. What are the implications?
* Yikes, the author is changing things in a PR labeled "Refactoring...". Now my guard is up and I have to review the other 500 lines in detail before I'm comfortable approving this. I need to ask, "That function that moved, did it change to? How can I tell?"

Reply:
We need a way to identify that a new node (validator) just joined the network and is voting on an old transaction. The vote will be discarded, but the node needs to be added to the list of nodes that are receiving broadcast (as part of avalanche). Otherwise the validator will not join the avalanche.

Maybe this is not the best way to solve it. Any suggestions on a better approach?

Reply:
That use case is certainly not what that error was intended for. I'd recommend creating an issue on the issue tracker describing this problem, discussing it either there or on Discord, and then creating a PR for that solution.

For this PR, your shortest path to getting it merged is to back out all the bits that change functionality. Reorging files and structs is large enough for one PR.

Reply:
Yes, please use separate PRs for separate efforts. We may want to merge either, neither or both. And those may be on different timelines. I'd hate to merge one before the other is ready because one is more urgent than the other.

Disabling the test sounds like the appropriate action because I think we'll need to think through the fix here.

For the errors, I'm considering going the opposite extreme, removing all data from all of them. For AccountNotFound, for example, that pubkey is only useful for debugging and we have better tools to expose that information. Code would never switch on the pubkey in an error object.

I'm also interested in voting solution that doesn't trigger AccountNotFound errors. Sounds like a terrible hack to begin with and one that we should be looking to purge, not build on.

Reply:
Cool! I agree. I'll rework the PR later today and push an update.